recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache License 2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        try:
+            if self.settings.arch == "armv8" and self.settings.os == "Android":
+                os_name = "Linux"
+            else:
+                os_name = str(self.settings.os)
+
+            archive_ext = "zip" if os_name == "Windows" else "tar.xz"
+            url = f"https://github.com/bytecodealliance/wasmtime/releases/download/v{self.version}/wasmtime-v{self.version}-{self.settings.arch}-{os_name.lower()}-c-api.{archive_ext}"
```

Comment:
This needs to be moved to the `conandata.yml` like this and you can add one for android

https://github.com/conan-io/conan-center-index/blob/9825ab5abcd595aed527c11725ae3ddaf145c8d6/recipes/nodejs/all/conandata.yml#L6

Reply:
Why ? It was packaged for Linux on CI successfully ...

Reply:
Such modification would require each time updating `conandata.yml`, my solution looks much simpler

Reply:
All downloads have to be in the `conandata.yml`, together with a sha256 checksum

Reply:
In CCI we aim to enable the most amount of workflows across a very diverse pool of consumers.

This is one of those items that's required and supported in all scenarios  this far

Reply:
@Croydon @prince-chrismc 
How to do it ???
`wasmtime` is compiled for different architectures due to this I was not able to add it in `conandata.yml` ...

Reply:
The example I provided above is very close... there are example for also involving `arch` which you can find (I would but mobile search is terrible )

Reply:
https://github.com/conan-io/conan-center-index/pull/6082

Reply:
Done